source/runtime_gui.cpp

```diff
@@ -2139,11 +2139,16 @@ void reshade::runtime::draw_gui_settings()
 		if (_back_buffer_format == reshade::api::format::r16g16b16a16_float ||
 			_back_buffer_color_space == reshade::api::color_space::hdr10_st2084)
 		{
-			modified |= ImGui::SliderInt(_("HDR PNG quality"), reinterpret_cast<int *>(&_screenshot_hdr_bits), 7, 16, "%d bit", ImGuiSliderFlags_AlwaysClamp);
+			// defaults to PNG to prevent empty selector when switching to HDR
+			if (_screenshot_format > 1) _screenshot_format = 0;
```

Comment:
This shouldn't overwrite the format unless the user explicitly changes it. Can do something like this instead:
```c++
int hdr_screenshot_format = _screenshot_format == 3 ? 1 : 0;
if (ImGui::Combo(_("Screenshot format"), reinterpret_cast<int *>(&hdr_screenshot_format), "Portable Network Graphics (*.png)\0JPEG XL Lossless (*.jxl)\0"))
{
	_screenshot_format = hdr_screenshot_format == 1 ? 3 : 1;
	modified = true;
}

if (hdr_screenshot_format == 0)
	modified |= ImGui::SliderInt(_("HDR PNG quality"), reinterpret_cast<int *>(&_screenshot_hdr_bits), 7, 16, "%d bit", ImGuiSliderFlags_AlwaysClamp);
```